core/drivers/versal_mbox.c

```diff
@@ -302,99 +428,109 @@ TEE_Result versal_mbox_alloc(size_t len, const void *init,
 	return TEE_SUCCESS;
 }
 
-TEE_Result versal_mbox_notify(struct versal_ipi_cmd *cmd,
+TEE_Result versal_mbox_free(struct versal_mbox_mem *mem)
+{
+	if (!mem)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	free(mem->buf);
```

Comment:
I strongly suggest to also set `mem->buf = NULL`

Reply:
Should be addressed by fixup commit [6d01a34](https://github.com/OP-TEE/optee_os/pull/6738/commits/6d01a341403ac150ba780df976a9c4903c3f435a)